src/main.cc

```diff
@@ -13,29 +14,97 @@ int main(int argc, char** argv) {
     //! IO
     std::unique_ptr<IO> io(new IO(argc, argv));
 
-    //! Mesh
-    std::unique_ptr<Mesh<3, 8>> mesh(new GMSH<3, 8>());
-
     //! MaterialProperties
-    std::shared_ptr<MaterialProperties> material =
-        std::make_shared<MaterialProperties>(io->material_properties());
+    std::vector<std::unique_ptr<MaterialProperties>> material;
+
+    for (unsigned i = 0; i < io->material_properties().size(); i++) {
+      material.emplace_back(
```

Comment:
Who owns the material?

Reply:
At this stage, `main` owns it but it is later moved to `mesh` and then `materialpoints`.
So in the end only `materialpoints` owns the `points` with its coordinates and ids, and the `materialproperties` too